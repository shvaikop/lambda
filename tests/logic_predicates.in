true = \x.\y.x
false = \x.\y.y
not = \p.p $false $true
or = \p.\q.p p q
and = \p.\q.p q p
zero = \f.\x.x
one = \f.\x.f x
iszero = \n.n (\x.$false) $true
$not $true
$and $true $false
$and $true $true
$or $true $false
$iszero $zero
$iszero $one
$or ($iszero $zero) ($iszero $one)
$and ($iszero $zero) ($iszero $one)
