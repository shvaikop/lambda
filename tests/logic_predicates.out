Welcome to Lambda Calculus Interpreter
\x.\y.y
\x.\y.y
\x.\y.x
\x.\y.x
\x.\y.x
\x.\y.y
\x.\y.x
\x.\y.y
